In [11]:
import sympy as sp # use symbolic python taking derivative
import handcalcs.render # render
from pint import UnitRegistry
u=UnitRegistry(auto_reduce_dimensions=True)
import numpy as np

In [16]:
%%render
d = 0.8*u.mm
L = (100+20)*u.mm
rho_f = 1000*u.kg/u.m**3
rho_m = 2.7*u.g/u.cm**3
g = 9.81*u.m/u.s**2

vol = np.pi*d**2/4*L
vol_immersed = 0.5*vol # immersed volume

F_w = rho_m*vol*g # gravity

F_b = rho_f*g*vol_immersed # buoyancy force
gamma = 72*u.mN/u.m
F_surface = gamma*L*2

<IPython.core.display.Latex object>

In [13]:
F_w.to(u.N)

0.001597658200620151 <Unit('newton')>

In [14]:
F_b.to(u.N)

0.00029586262974447237 <Unit('newton')>

In [17]:
F_surface.to(u.N)

0.01728 <Unit('newton')>

# conclusion
the buoyancy force is 0.0002959 N which is not bigger enough to balance the weight 0.001598 N. on the other hand, the surface tension force can be as big as 0.01728 N. So it is the surface tension to balance the weight. Of course, the buoyancy force played a minor role. 